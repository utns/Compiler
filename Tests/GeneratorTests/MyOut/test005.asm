global  main
extern  printf
section .text
main:
	mov rax, 2
	mov rbx, 3
	mov rsi, rax
	add rsi, rbx
	xor rax, rax
	mov rdi, formatInt
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
	mov rax, 2
	mov rbx, 3
	mov rsi, rax
	sub rsi, rbx
	xor rax, rax
	mov rdi, formatInt
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
	mov rsi, 60
	xor rax, rax
	mov rdi, formatInt
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
	mov rax, 20
	mov rbx, 3
	xor rdx, rdx
	idiv rbx
	mov rsi, rax
	xor rax, rax
	mov rdi, formatInt
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
	mov rax, 20
	mov rbx, 3
	xor rdx, rdx
	idiv rbx
	mov rsi, rdx
	xor rax, rax
	mov rdi, formatInt
	sub rsp, 8
	call printf
	add rsp, 8
	xor rax, rax
	mov rdi, formatNewLine
	sub rsp, 8
	call printf
	add rsp, 8
section .data
	formatInt: db "%ld", 0
	formatFloat: db "%f", 0
	formatNewLine: db 10, 0
