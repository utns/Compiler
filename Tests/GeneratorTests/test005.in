begin
    writeln(2 + 3);
    writeln(2 - 3);
    writeln(20 * 3);
    writeln(20 div 3); 
    writeln(20 mod 3);
end.