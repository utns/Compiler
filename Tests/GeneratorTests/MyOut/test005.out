5
-1
60
6
2
